packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -0,0 +1,316 @@
+      logStreamProvider: logStreamProvider,
+      logStore: logStore,
+    );
+    if (plugin._pluginConfig.flushIntervalInSeconds > Duration.zero) {
+      plugin._timer = StoppableTimer(
+        duration: plugin._pluginConfig.flushIntervalInSeconds,
+        callback: plugin._startSyncingIfNotInProgress,
+        onError: plugin._onTimerError,
+      );
+    }
+    return plugin;
+  }
+
+  /// An [AWSLoggerPlugin] to use only for testing.
+  @protected
+  @visibleForTesting
+  CloudWatchLoggerPlugin.testPlugin({
+    required CloudWatchLogsClient client,
+    required CloudWatchLoggerPluginConfiguration pluginConfig,
+    required CloudWatchLogStreamProvider logStreamProvider,
+    required QueuedItemStore logStore,
+    RemoteLoggingConstraintProvider? remoteLoggingConstraintProvider,
+  })  : _syncing = false,
+        _enabled = pluginConfig.enable,
+        _pluginConfig = pluginConfig,
+        _logStore = logStore,
+        _remoteLoggingConstraintProvider = remoteLoggingConstraintProvider,
+        _logStreamProvider = logStreamProvider,
+        _clinet = client;
+
+  CloudWatchLoggerPlugin._({
+    required AWSCredentialsProvider credentialsProvider,
+    required CloudWatchLoggerPluginConfiguration pluginConfig,
+    RemoteLoggingConstraintProvider? remoteLoggingConstraintProvider,
+    CloudWatchLogStreamProvider? logStreamProvider,
+    QueuedItemStore? logStore,
+  })  : _syncing = false,
+        _enabled = pluginConfig.enable,
+        _pluginConfig = pluginConfig,
+        _logStore = logStore ?? InMemoryQueuedItemStore(),
+        _remoteLoggingConstraintProvider = remoteLoggingConstraintProvider ??
+            (pluginConfig.defaultRemoteConfiguration != null
+                ? DefaultRemoteLoggingConstraintProvider(
+                    config: pluginConfig.defaultRemoteConfiguration!,
+                    credentialsProvider: credentialsProvider,
+                  )
+                : null),
+        _clinet = CloudWatchLogsClient(
+          region: pluginConfig.region,
+          credentialsProvider: credentialsProvider,
+        ),
+        _logStreamProvider = logStreamProvider ??
+            DefaultCloudWatchLogStreamProvider(
+              logGroupName: pluginConfig.logGroupName,
+              client: CloudWatchLogsClient(
+                region: pluginConfig.region,
+                credentialsProvider: credentialsProvider,
+              ),
+            );
+
+  final CloudWatchLoggerPluginConfiguration _pluginConfig;
+  final CloudWatchLogsClient _clinet;
+  final CloudWatchLogStreamProvider _logStreamProvider;
+  final QueuedItemStore _logStore;
+  bool _syncing;
+  bool _enabled;
+  StoppableTimer? _timer;
+  RemoteLoggingConstraintProvider? _remoteLoggingConstraintProvider;
+
+  set remoteLoggingConstraintProvider(
+    RemoteLoggingConstraintProvider remoteProvider,
+  ) {
+    if (_remoteLoggingConstraintProvider != null) {
+      logger.error(
```

Comment:
it can throw, the reason for not throwing is the plugin can function and so logging error might be sufficient. tbh I'm not sure which one is a better in this case.

Reply:
If a developer calls this, they would expect it to set the provider. A log message could easily get lost. Opt for throwing.